Purchases/Misc/Obsoletions.swift

```diff
@@ -489,11 +489,118 @@ public extension StoreProductDiscount.PaymentMode {
+    @available(watchOS, obsoleted: 1)
+    @available(macOS, obsoleted: 1)
+    @available(macCatalyst, obsoleted: 1)
+    enum ReadableErrorCodeKey {}
+
+    @available(iOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(tvOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(watchOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macCatalyst, obsoleted: 1, message: "Remove `Purchases.`")
+    enum ErrorCode {}
+
+    @available(iOS, obsoleted: 1)
+    @available(tvOS, obsoleted: 1)
+    @available(watchOS, obsoleted: 1)
+    @available(macOS, obsoleted: 1)
+    @available(macCatalyst, obsoleted: 1)
+    enum RevenueCatBackendErrorCode {}
+
+    @available(iOS, obsoleted: 1, renamed: "StoreTransaction")
+    @available(tvOS, obsoleted: 1, renamed: "StoreTransaction")
+    @available(watchOS, obsoleted: 1, renamed: "StoreTransaction")
+    @available(macOS, obsoleted: 1, renamed: "StoreTransaction")
+    @available(macCatalyst, obsoleted: 1, renamed: "StoreTransaction")
+    enum Transaction {}
+
+    @available(iOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(tvOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(watchOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macCatalyst, obsoleted: 1, message: "Remove `Purchases.`")
+    enum EntitlementInfo {}
+
+    @available(iOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(tvOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(watchOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macCatalyst, obsoleted: 1, message: "Remove `Purchases.`")
+    enum EntitlementInfos {}
+
+    @available(iOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(tvOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(watchOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macCatalyst, obsoleted: 1, message: "Remove `Purchases.`")
+    enum PackageType {}
+
+    @available(iOS, obsoleted: 1, renamed: "CustomerInfo")
+    @available(tvOS, obsoleted: 1, renamed: "CustomerInfo")
+    @available(watchOS, obsoleted: 1, renamed: "CustomerInfo")
+    @available(macOS, obsoleted: 1, renamed: "CustomerInfo")
+    @available(macCatalyst, obsoleted: 1, renamed: "CustomerInfo")
+    enum PurchaserInfo {}
+
+    @available(iOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(tvOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(watchOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macOS, obsoleted: 1, message: "Remove `Purchases.`")
+    @available(macCatalyst, obsoleted: 1, message: "Remove `Purchases.`")
+    enum Offering {}
```

Comment:
I thought these would cause conflicts in `Purchases.swift`? I guess I misunderstood the issue

Reply:
This one must not be referenced from inside an extension cause it doesn't give any errors.

Reply:
ohhh, so the issue only happens if a type is referenced in an _extension_ of `Purchases`? I guess this one is safe but most of the others are referenced from `Purchases+async`, it's just that we don't have a public method that gives you a single `Offering` but rather `Offerings`